ext/mysqli/mysqli_api.c

```diff
@@ -970,7 +970,7 @@ void mysqli_stmt_fetch_libmysql(INTERNAL_FUNCTION_PARAMETERS)
 			zval *result;
 			/* it must be a reference, isn't it? */
 			if (Z_ISREF(stmt->result.vars[i])) {
-				result = stmt->result.vars[i];
+				*result = stmt->result.vars[i];
```

Comment:
I think this should be `result = &stmt->result.vars[i]`. It doesn't really make sense to do a copy here.